src/gui/tray/UnifiedSearchResultItem.qml

```diff
@@ -0,0 +1,232 @@
+import QtQml 2.12
+import QtQuick 2.9
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.2
+import Style 1.0
+import QtGraphicalEffects 1.0
+
+MouseArea {
+    id: unifiedSearchResultMouseArea
+
+    property int defaultHeight: 0
+
+    readonly property int iconLeftMargin: 12
+    readonly property int textLeftMargin: 4
+    readonly property int textRightMargin: 16
+    readonly property int categorySeparatorLeftMargin: 16
+
+    readonly property bool isFetchMoreTrigger: model.typeAsString === "FetchMoreTrigger"
+    readonly property bool isCategorySeparator: model.typeAsString === "CategorySeparator"
+    readonly property bool isDefaultItem: model.typeAsString === "Default"
+
+    property bool isFetchMoreInProgress: unifiedSearchResultsModel.currentFetchMoreInProgressProviderId === model.providerId
+    property bool isSearchInProgress: false
+
+    enabled: !isCategorySeparator && (!isFetchMoreTrigger || !isSearchInProgress)
+    hoverEnabled: !isCategorySeparator && (!isFetchMoreTrigger || !isSearchInProgress)
+
+    height: !isCategorySeparator ? defaultHeight : defaultHeight / 2
+
+
+    onClicked: {
+        if (isFetchMoreTrigger) {
+            unifiedSearchResultsModel.fetchMoreTriggerClicked(model.providerId)
+        } else if (isDefaultItem) {
+            unifiedSearchResultsModel.resultClicked(model.providerId, model.resourceUrlRole)
+        }
+    }
+
+    ToolTip {
+        parent: unifiedSearchResultMouseArea
+        visible: !unifiedSearchResultMouseArea.isCategorySeparator && unifiedSearchResultMouseArea.containsMouse
+        text: isFetchMoreTrigger ? qsTr("Load more results") : model.resultTitle + "\n\n" + model.subline
+        delay: 1000
+    }
+
+    Rectangle {
+        id: unifiedSearchResultHoverBackground
+        anchors.fill: parent
+        color: (parent.containsMouse ? Style.lightHover : "transparent")
+    }
+
+    RowLayout {
```

Comment:
as a more general comment on this item
you are in reality defining 3 different mutually exclusive items inside a single one
this is not good because you waste resources and this is more complex than needed
please move the definition of the 3 `RowLayout` to separate files and instantiate them here using `Loader` such that only one is ever created
```
Loader {
    active: !isFetchMoreTrigger && !isCategorySeparator
    sourceComponent: SearchResultItem {
        [...]
    }
}
Loader {
    active: isFetchMoreTrigger
    sourceComponent: FetchMoreItem {
        [...]
    }
}
[...]
```